File: Backend/exam-server/src/main/java/com/exam/examserver/service/QuizService.java
package com.exam.examserver.service;

import com.exam.examserver.entity.Category;
import com.exam.examserver.entity.Quiz;

import java.util.List;

public interface QuizService {

    Quiz addQuiz(Quiz quiz);

    Quiz updateQuiz(Quiz quiz);

    List<Quiz> getAllQuizzes();

    Quiz getByQuizId(long qId);

    void deleteByQuizId(long qId);

    List<Quiz> getQuizzesByCategoryId(long cId);

    List<Quiz> getAllActiveQuizzes();

    List<Quiz> getActiveQuizzesByCategoryId(long cId);
}

File: Frontend/exam-portal-fe/src/app/services/quiz.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import baseUrl from './helper';

@Injectable({
  providedIn: 'root'
})
export class QuizService {

  constructor(private _httpClient: HttpClient) { }

  public getAllQuizzes() {
    return this._httpClient.get(`${baseUrl}/api/quizzes`);
  }

  public addQuiz(quiz: any) {
    return this._httpClient.post(`${baseUrl}/api/quizzes`, quiz);
  }

  public deleteQuiz(qid: any) {
    return this._httpClient.delete(`${baseUrl}/api/quizzes/${qid}`);
  }

  public getQuizById(qid: any) {
    return this._httpClient.get(`${baseUrl}/api/quizzes/${qid}`);
  }

  public updateQuiz(quiz: any) {
    return this._httpClient.put(`${baseUrl}/api/quizzes`, quiz);
  }

  public getAllActiveQuizzes() {
    return this._httpClient.get(`${baseUrl}/api/quizzes/active`);
  }

  public getAllActiveQuizzesByCategoryId(cid: any) {
    return this._httpClient.get(`${baseUrl}/api/quizzes/active/category/${cid}`)
  }
}
